presentation/src/main/java/dominando/android/presentation/BookFormViewModel.kt

```diff
@@ -21,34 +24,62 @@ class BookFormViewModel(
     private val saveBookUseCase: SaveBookUseCase
 ) : ViewModel(), LifecycleObserver {
 
-    var book: BookBinding? = null
     var tempImageFile: File? = null
         set(value) {
             deleteTempPhoto()
             field = value
             shouldDeleteImage = true
         }
+
+    val publishers = listOf(
+            Publisher("1", "Novatec"),
+            Publisher("2", "Outra")
+    )
+
     private var shouldDeleteImage: Boolean = true
 
-    private val state: MutableLiveData<LiveEvent<ViewState<Unit>>> = MutableLiveData()
+    private val saveBookEvent: MutableLiveData<LiveEvent<ViewState<Unit>>> = MutableLiveData()
+    private val book = MutableLiveData<BookBinding>(BookBinding())
+
+    val state = Transformations.map(saveBookEvent) {
+        /* Retrieve data before call consumeEvent() */
+        val data = it.peekContent()
+        if (it.peekContent()?.status == ViewState.Status.ERROR) {
+            it.consumeEvent()
+        }
+        data
+    }
 
-    fun getState(): LiveData<LiveEvent<ViewState<Unit>>> = state
+    fun book(): LiveData<BookBinding> = book
 
-    fun saveBook(book: BookBinding) {
-        state.postValue(LiveEvent(ViewState(ViewState.Status.LOADING)))
+    fun saveBook() {
+        saveBookEvent.postValue(LiveEvent(ViewState(ViewState.Status.LOADING)))
         viewModelScope.launch {
             try {
                 withContext(Dispatchers.IO) {
-                    saveBookUseCase.execute(BookConverter.toData(book))
+                    book.value?.let { saveBookUseCase.execute(BookConverter.toData(it)) }
```

Comment:
If `book` is null, it will do a false successful save operation. We must must first validate if the book is not null.

Reply:
Fixed up